OptimizelySDKCore/OptimizelySDKCore/OPTLYMacros.h

```diff
@@ -15,3 +15,7 @@
  ***************************************************************************/
 
 #define StringOrEmpty(A)  ({ __typeof__(A) __a = (A); __a ? __a : @""; })
+
+#define isEmptyString(A)  ({ !(A && [A respondsToSelector:@selector(isEqualToString:)] && ![(NSString *)A isEqualToString:@""]); })
+
+#define isEmptyArray(A)  ({ !(A && [A respondsToSelector:@selector(count)] && [(NSArray *)A count] > 0); })
```

Comment:
I would like to propose that we get rid of the OPTLYMacros.h file entirely.

First, the macros I see in OPTLYMacros.h are sparingly used:
* isEmptyString only used by OPTLYDecisionService.m in 1 place.
* isEmptyArray only used by OPTLYDecisionService.m in 2 places.
* Only other entry in OPTLYMacros.h is the preexisting StringOrEmpty which does show up as more occurrences but in only 1 file: OPTLYEventBuilder.m .

Second, there disadvantages to using macros instead of code when an alternative
code solution is available.  Code is much easier to debug at run time and more easily
understood.

I'd like to suggest adding a "stringOrEmpty:" method to OPTLYEventBuilder.m and
"isEmptyString:" and "isEmptyArray:" methods to OPTLYDecisionService.m .  My
comment can be taken as asking you to code these 3 methods, use these 3 methods
instead of the macros, and delete OPTLYMacros.h from the workspace.  These 3 methods
could be "static" "+" methods in those 2 classes.